api/src/main/resources/plugins/base_common.py

```diff
@@ -21,29 +21,79 @@
-        self._control_component(create_data['stop_cmds'])
+        # self._control_component(create_data['stop_cmds'])
 
     def _control_component(self, cmds):
-        key_file = self._environment['cluster_private_key']
+        '''key_file = self._environment['cluster_private_key']
         root_user = self._environment['cluster_root_user']
-        target_host = 'localhost'
-        deployer_utils.exec_ssh(target_host, root_user, key_file, cmds)
+        target_host = 'localhost'''
+        deployer_utils.exec_cmds(cmds)
+
+    def create_custom_resource_object(self, spark_app_json_path):
+
+        config.load_incluster_config()
+        configuration = client.Configuration()
+        api_instance = client.CustomObjectsApi(client.ApiClient(configuration))
+        group = 'sparkoperator.k8s.io'  # str | The custom resource's group name
+        version = 'v1beta2'  # str | The custom resource's version
+        namespace = 'pnda'  # str | The custom resource's namespace
+        plural = 'sparkapplications'  # str | The custom resource's plural name. For TPRs this would be lowercase plural kind.
+        pretty = 'true'
+        with open(spark_app_json_path, 'r') as jfile:
+
+            data = json.load(jfile)
+
+        try:
+            api_response = api_instance.create_namespaced_custom_object(group, version, namespace, plural, body=data, pretty=pretty)
+            logging.info("CRD create response %s" % api_response)
+        except ApiException as e:
+            logging.debug("Exception when calling CustomObjectsApi->create_namespaced_custom_object: %s\n" % e)
+
+    def delete_custom_resource_object(self, application_name):
+        config.load_incluster_config()
+        configuration = client.Configuration()
+        api_instance = client.CustomObjectsApi(client.ApiClient(configuration))
+        group = 'sparkoperator.k8s.io'  # str | The custom resource's group name
+        version = 'v1beta2'  # str | The custom resource's version
+        namespace = 'pnda'  # str | The custom resource's namespace
+        plural = 'sparkapplications'  # str | The custom resource's plural name. For TPRs this would be lowercase plural kind.
+        name = application_name  # str | the custom object's name
+        body = client.V1DeleteOptions()
+        grace_period_seconds = 56
+        orphan_dependents = True
+
+        try:
+            api_response = api_instance.delete_namespaced_custom_object(
+                group,
+                version,
+                namespace,
+                plural,
+                name,
+                body,
+                grace_period_seconds=grace_period_seconds,
+                orphan_dependents=orphan_dependents)
+            logging.info("CRD delete response %s" % api_response)
+        except ApiException as e:
+            logging.debug("Exception when calling CustomObjectsApi->delete_namespaced_custom_object: %s\n" % e)
```

Comment:
This is code specific to sparkStreaming in Common class used by Spark and Flink plugins. 
Must be moved to specific sparkStreaming.py code.

Reply:
Hi Carlos,

I have moved spark streaming specific code into sparkStreaming.py and tested successfully. Please review and let us know.

Regards,
-Sreenivasa